### DISCUSSION TITLE
How to run intro.lua?

### DISCUSSION THREAD

**Discussion by Proful** [grapp-dev/nui-components.nvim#2] [Q&A] [answered]
After Lazy setup, where to keep & how to trigger it to run
https://gist.github.com/mobily/cfaa4264f1094e20e25f31f44de6c21f

**Comment by mehalter** [comment, marked as answer]
You can put this really anywhere where you want to build a component and run it. A good test would be to just put it in your `~/.config/nvim/lua` folder so say you put it at `~/.config/nvim/lua/nui_components_intro.lua` then when you run neovim you would simply run `:lua require("nui_components_intro")` and it would execute the code.

That require just loads that module and run the code inside of it. If you have a function somewhere else like attached to a key binding or something where you want to build a component you can do it there or organize it somewhere else in your config and then `require` it

**Comment by mehalter** [comment]
One thing to note is that gist is missing a couple imports, so just make sure to add this at the top of the file:

```lua
local n = require "nui-components"
local spinner_formats = require "nui-components.utils.spinner-formats"
```

**Comment by mobily** [comment]
@mehalter thanks for the detailed explanation! I have just updated the gist content :)
@Proful if this is what you need, then mark Micah's post as the answer, thanks!

**Reply by mehalter** [reply]
No problem! I was just trolling around GitHub and stumbled across the issue and figured I would help out 😅 awesome work with this cool API!
